lib/sdotypes.c

```diff
@@ -1934,27 +1856,32 @@ void SDODNSEmpty(sdo_dns_name_t *b)
 char *sdo_read_dns(sdor_t *sdor)
 {
 	char *buf;
-	int len;
+	size_t len;
 
 	/* read length of DNS */
-	len = sdo_read_string_sz(sdor);
+	if (sdor_string_length(sdor, &len) || len <= 0) {
+		// TO-DO : Evaluate this
+		// sdo_free(buf);
+		return NULL;
+	}
 
 	buf = sdo_alloc(len + 1);
 
 	if (!buf)
 		return NULL;
 
-	sdo_read_string(sdor, buf, len + 1);
-
-	if (len == 0) {
+	if(!sdor_text_string(sdor, buf, len + 1)) {
```

Comment:
Is there a way to do without the "+ 1"? Too easy to forget or make a mistake.

Reply:
Would be fixed in a subsequent PR.